src/web/mjs/connectors/Sukima.mjs

```diff
@@ -0,0 +1,191 @@
+        super.id = 'sukima';
+        super.label = 'Sukima';
+        this.tags = [ 'manga', 'japanese' ];
+        this.url = 'https://www.sukima.me';
+        this.language = 'japanese';
+        this.requestOptions.headers.set( 'x-referer', this.url );
+        this.lock = false;
+    }
+
+    async _getMangas() {
+        let request = new Request(new URL('/api/book/v1/free/', this.url), {
+            method: 'POST',
+            body: '{"store":false,"genre":"0"}',
+            headers: {
+                'Content-Type': 'application/json;charset=utf-8'
+            }
+        });
+        let categorys = await this.fetchJSON(request);
+
+        let uri;
+        let pages;
+        let mangas = [];
+        for (const category of categorys.rows) {
+            if (category.more_btn.link.startsWith('/book/search/')) {
+                uri = new URL(category.more_btn.link, this.url);
+                pages = 1;
+                for (let page = 1; page <= pages; page++) {
+                    request = new Request(new URL('/api/v1/search/', this.url), {
+                        method: 'POST',
+                        body: '{"free":["1","2"],"page":'+page+',"sort_by":"0","tag":["'+uri.searchParams.get('tag')+'"]}',
+                        headers: {
+                            'Content-Type': 'application/json;charset=utf-8',
+                        }
+                    });
+
+                    let pageContent = await this.fetchJSON(request);
+                    pages = pageContent.max_page;
+
+                    for (const manga of pageContent.items) {
+                        mangas.push(
+                            {
+                                id: '/api/book/v1/title/'+manga.title_code+'/',
+                                title: manga.title_name
+                            }
+                        );
+                    }
+                }
+            }
+        }
+
+        return mangas;
+    }
+
+    async _getChapters(manga) {
+        let request = new Request(new URL(manga.id, this.url), {
+            method: 'POST',
+            body: '{}',
+            headers: {
+                'Content-Type': 'application/json;charset=utf-8',
+            }
+        });
+
+        let chapters = [];
+        let books = await this.fetchJSON(request);
+        for (const book of books.contents) {
+            for (const chapter of book.stories) {
+                chapters.push(
+                    {
+                        id: '/bv/t/' + chapter.title_code + '/v/' + chapter.volume + '/s/' + chapter.story + '/p/0',
+                        title: chapter.volume.toString().padStart(3, '0') + '-' + chapter.story.toString().padStart(3, '0') + ' - ' + chapter.info.text.trim()
+                    }
+                );
+            }
+        }
+        return chapters;
+    }
+
+    async _getPages(manga) {
+        let script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
```

Comment:
Errors in `setTimeout` are not captured by the `Promise`, so i suggested to encapsulate the whole body of the function within the `setTimeout` in a try...catch block where the error does properly reject the promise, otherwise the Promise may never resolve/reject and HakuNeko hangs
```javascript
setTimeout(() => {
    try {
        // ...
    } catch(error) {
        reject(error);
    }
}, 2500)
```

Reply:
But I believe then I'll not receive a Promise which I can `await`. This would then be promise(settimeout(promise())) which feels terribly wrong. Do I miss something?

Reply:
This is how it should look, so the promise is save and will either resolve or reject:
```javascript
new Promise((resolve, reject) => {
    setTimeout(() => {
        try {
            // simulate an unexpected exception ...
            throw new Error('Test the promise rejects correctly ...');
            // resolve the pending promise
            resolve('OK');
        } catch(error) {
            // reject the pending promise
            reject(error);
        }
    }, 2500);
});
```
Without the try-catch block, the `Promise` will be pending forever, because `resolve('OK')` is never reached and the exception is thrown outside of the promise itself, because `setTimeout` is called asynchronous.

Without an asynchronous call, there is no need for a try-catch block, because the Promise will correctly capture and reject any exception automatically:

```javascript
new Promise(resolve => {
    // simulate an unexpected exception ...
    throw new Error('Test the promise rejects correctly ...');
    // => the exception is captured and the promise rejects automatically
    resolve('OK');
});
```

Reply:
Got it! Thanks.